### ISSUE TITLE
extract_from_sequences.py script run problem

### ISSUE THREAD

**Issue by summerxiajq** [bcgsc/Terminitor#2]
The output file test extracted by the extract_from_sequences.py script is empty, is it because the reference file I provided is wrong? The command :./src/extract_from_sequences.py -t /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.gtf -a /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.chr.gtf -g /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa -m aln4.bam -o test

**Comment by cheny19** [comment]
Hi @summerxiajq ,

First of all, could you double-check how you generated aln4.bam? Is it aligned against the correct genome assembly?

Second, how did you get `/data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.gtf` this file? You should use `awk '$3=="transcript"' /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.chr.gtf ` to get only transcript annotations. 

If both are fine, maybe it's better to send me your test file so I can take a closer look.

Thanks,
Chen

**Comment by summerxiajq** [comment]
Good morning, teacher:
 
Thank you very much for your quick reply. I have read your article very well. What I understand is that your method can predict the new PAS, even if it is in the intron region.I have corrected it in your way, but I still can't get the output. Here's how I did it:
 
&nbsp;
 
At first, I didn't use the parameter &nbsp;-stranded. Later, I thought that my data was chain specific, so I added this parameter. Of course, I did the following analysis for both versions of the resulting bam
 
rnabloom -stranded -left /data1/zhoulab/xiajiaqi/IPA/3seq_RNAseq/SRR6830272_2.fastq.gz -right /data1/zhoulab/xiajiaqi/IPA/3seq_RNAseq/SRR6830272_1.fastq.gz -revcomp-right -outdir assembly -a 4 -e 1 -stratum 01 -ss -ntcard -fpr 0.005
 
&nbsp;
 
Is the reference sequence fasta I provided for this step correct?
 
minimap2 -ax splice /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.cdna.all.fa /data1/zhoulab/xiajiaqi/assembly/rnabloom.transcripts.fa | samtools view -b - &gt; ../aln.bam
 
Added the command you provided to me.
 
awk '$3=="transcript"' /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.chr.gtf &gt;Homo_sapiens.GRCh38.99.transcripts.gtf
 
&nbsp;
 
Homo_sapiens.GRCh38.dna.primary_assembly.fa downloaded it directly from the website.
 
&nbsp;
 
./src/extract_from_sequences.py -t /data1/zhoulab/xiajiaqi/reference/hg38/ Homo_sapiens.GRCh38.99.transcripts.gtf -a /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.chr.gtf -g /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.dna.primary_assembly.fa -m ../aln.bam -o test
 
&nbsp;
 
I would like to give you my fastq, but the file is too large. Is it ok if I provide you with the file rnabloom.transcripts. fa?I hope I can successfully use your method.

  

 

 ------------------&nbsp;原始邮件&nbsp;------------------
  发件人:&nbsp;"Chen Yang"<notifications@github.com&gt;;
 发送时间:&nbsp;2020年2月29日(星期六) 凌晨3:34
 收件人:&nbsp;"bcgsc/Terminitor"<Terminitor@noreply.github.com&gt;;
 抄送:&nbsp;"夏佳琪"<1360983892@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
 主题:&nbsp;Re: [bcgsc/Terminitor] extract_from_sequences.py script run problem (#2)

 

 
Hi @summerxiajq ,
 
First of all, could you double-check how you generated aln4.bam? Is it aligned against the correct genome assembly?
 
Second, how did you get /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.gtf this file? You should use awk '$3=="transcript"' /data1/zhoulab/xiajiaqi/reference/hg38/Homo_sapiens.GRCh38.99.chr.gtf to get only transcript annotations.
 
If both are fine, maybe it's better to send me your test file so I can take a closer look.
 
Thanks,
Chen

**Comment by cheny19** [comment]
Yes, if your RNA-Seq data is strand-specific, then you should use `-stranded` when running RNA-Bloom.

When you are aligning assembled transcripts to the reference, you should use reference genome instead of cdna fasta file. I thing this is the main reason why there's no output.

**Comment by summerxiajq** [comment]
Hello:

&nbsp; &nbsp; &nbsp;Excuse me.I got the output and integrated it with the site as follows: &gt; 3.l.1_chr1_149530108_f 0.01623591.I'm not sure what the number 0.01623591 means.How likely is this site to be the ployA site, or the p-value?How should this threshold be?The article reads:Ntor Termin (A) takes the fixed length sequence as input and performs three - label classification to determine been the sequence contains A poly real site (A), A non - polyadenylated CS,From the result, how can I distinguish the three kinds of sites?Is the error of 30bp above and below this site acceptable?Is the fasta in the result file based on the expression quantity to get the sequence?Can the fasta only be in 3’utr region?Does this approach fully consider the PloyA site in the intron region?Sorry to disturb you again, because the following processing needs the above information, hope to get your help, wish you good luck.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Chen Yang"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月3日(星期二) 凌晨2:24
收件人:&nbsp;"bcgsc/Terminitor"<Terminitor@noreply.github.com&gt;;
抄送:&nbsp;"夏佳琪"<1360983892@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [bcgsc/Terminitor] extract_from_sequences.py script run problem (#2)




Yes, if your RNA-Seq data is strand-specific, then you should use -stranded when running RNA-Bloom.
 
When you are aligning assembled transcripts to the reference, you should use reference genome instead of cdna fasta file. I thing this is the main reason why there's no output.

**Comment by cheny19** [comment]
The number is the probability of this specific site being a poly(A) site, normally the threshold is 0.5. Based on our experience, if the real site is ~30bp within the vicinity, the probability should still be pretty, but not necessary. The fasta file is purely based on assembly, which is not dependent on expression level. Therefore, it's not only in the 3'utr region either. If the terminal exon is short enough, the extracted sequence may contain more upstream regions as well. If the poly(A) site occurs in a intron region and it is expressed, which means that intron is used as terminal exon for that transcript, so yes, it will be considered in our program, because we only rely on the transcript reconstruction, not the intron/exon information.

Hope that explains your questions. Feel free to contact and I'm happy to help.
